lsst/faro
# DM-26214: Switch to using sourceTable_visit parquet files instead of src FITS files

This PR adds base classes that use `sourceTable_visit` parquet files as input to per-visit and per-detector metrics.

To test, run a command like

`pipetask --long-log run -b /repo/main/butler.yaml --register-dataset-types -p testpipe.yaml -d "visit=35892 AND skymap='hsc_rings_v1' AND instrument='HSC'" --output u/kbechtol/sourcetable_test -i HSC/runs/RC2/w_2021_18/DM-29973 --timeout 999999`

With testpipe.yaml pipeline file something like

```
description: Compute metrics from sourceTable_visit catalogs
tasks:
  nsrcMeasVisitTable:
    class: lsst.faro.measurement.VisitTableMeasurementTask
    config:
      connections.package: info
      connections.metric: nsrcMeasVisitTable
      python: |
        from lsst.faro.base import NumSourcesTask
        config.measure.retarget(NumSourcesTask)
        config.columns = 'coord_ra, coord_dec, visit'

description: Compute per-detector metrics from sourceTable_visit catalogs
tasks:
  nsrcMeasDetectorTable:
    class: lsst.faro.measurement.DetectorTableMeasurementTask
    config:
      connections.package: info
      connections.metric: nsrcMeasDetectorTable
      python: |
        from lsst.faro.base import NumSourcesTask
        config.measure.retarget(NumSourcesTask)
        config.columns = 'coord_ra, coord_dec, visit, detector'
```

In the current PR version, there are new base classes that use `sourceTable_visit` rather than replacing the previous base classes that use `src` catalogs, i.e., we retain both versions for now. Perhaps we should discuss at what point / if we want to completely replace the versions that use `src` catalogs. Another question for this PR is whether we want to add base classes for the object tables and for creating the matching catalogs, or whether we want those as separate PRs.

Currently, this PR does not have new/modified pipeline yaml files included.
